evap/contributor/views.py

```diff
@@ -17,17 +17,20 @@
 @contributor_or_delegate_required
 def index(request):
     user = request.user
+    show_delegated = get_parameter_from_url_or_session(request, "show_delegated", True)
 
     contributor_visible_states = ['prepared', 'editor_approved', 'approved', 'in_evaluation', 'evaluated', 'reviewed', 'published']
     own_courses = Course.objects.filter(contributions__contributor=user, state__in=contributor_visible_states)
 
     represented_users = user.represented_users.all()
     delegated_courses = Course.objects.exclude(id__in=own_courses).filter(contributions__can_edit=True, contributions__contributor__in=represented_users, state__in=contributor_visible_states)
 
-    all_courses = list(own_courses) + list(delegated_courses)
-    all_courses.sort(key=lambda course: list(STATES_ORDERED.keys()).index(course.state))
+    displayed_courses = list(own_courses)
+    if show_delegated:
+        displayed_courses += list(delegated_courses)
```

Comment:
right now, if the user chose to not show delegated courses, you are still getting them only to check in the template whether any such course exists.
i'd move the `delegated_courses` here and check whether the user `is_delegate` in the template.

Reply:
done and i completely removed the delegated courses from the template data by adding an attribute to the delegated courses.